### DISCUSSION TITLE
Reduce the computational time for simulation long cycles

### DISCUSSION THREAD

**Discussion by kumaryash7** [pybamm-team/PyBaMM#3574] [Q&A]
I am performing degradation analysis for a long number of cycles in many iterations. It takes a lot of computational time and also memory usage is high.
How to optimise the code to reduce the computational time?
Thanks in advance.

**Comment by martinjrobins** [comment]
for reducing your memory usage, you could look into the `output_variables` parameter for the IDAKLU solver (see https://docs.pybamm.org/en/latest/source/api/solvers/idaklu_solver.html). This will only return the output variables you are interested in rather than the entire solution

**Reply by kumaryash7** [reply]
Thank you, 
Can you suggest some techniques to reduce computational time
